Information Architecture Refactor: Update the Plan > Lesson Summary

AlexVelezLl opened the pull request:
## Summary
Update the Plan > Lesson Summary page to reflect both planning and report elements.




## References
Closes #12707


## Reviewer guidance
* Does the coach plan lessons follows the behaviour described in the [specs](https://www.figma.com/design/lVJt5ukOrxS9qay0rXy7GC/0.18-Coach-updates?node-id=94-37724&node-type=frame&t=p3Mj0LPbjVUbeCnS-0)?


## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

AlexVelezLl committed:
Check tableHasRows for transition-groups

AlexVelezLl committed:
Make report lessons resources table editable

AlexVelezLl committed:
Rename ReportsLessonResourcesTable

AlexVelezLl committed:
Introduce lessonResourcesTable in lessonSummaryPage

AlexVelezLl committed:
Add functionality to sort and delete resources

AlexVelezLl committed:
Fix resources backup

AlexVelezLl committed:
ExportCSV for lesson resources table

AlexVelezLl committed:
Add learners table

AlexVelezLl committed:
Modify lesson status and reorganize lesson header

AlexVelezLl committed:
Fix text selection

marcellamaki commented:
@radinamatic @pcenov - if we can get QA review here, that would be great. Peter, the focus can be on overall alignment to the spec. Radina, a focus on any accessibility regressions in particular would be helpful! Thank you :)

AlexVelezLl committed:
Avoid reload on tab change and show more options in dropdown menu

pcenov commented:
Hi @AlexVelezLl, I was able to identify the following minor differences with the spec:

1. The title of the lesson is not centered and it should be checked whether the icon and text size are ok.
2. I don't see the 'Made visible {date} ago' text - is this part of a separate PR?
3. The position of the header row of the table and also the missing underline.
4. The spacing between the labels and the text in the sidebar is not the same as in the Figma frame.

![lesson summary](https://github.com/user-attachments/assets/298e572b-2169-4c8c-b02e-7b69a6170ac4)

AlexVelezLl commented:
Thank you @pcenov! 
1. Let me check this.
2. Let me check this with the team.
3. I think this line can be ignored for now, as this is part of the current table and I think updating this style is out of scope for now, right @marcellamaki?
4. Will fix this!

AlexVelezLl committed:
Fix styles

AlexVelezLl commented:
Hi @pcenov I have fixed the point 1, and point 4. I have fixed the alignment between the title and the actions, and fixed the font size.

I discussed with the team and they said that point 2 can be managed in a follow up issue to further discuss the implications of introducing it.

marcellamaki commented:
Yes - thank you @AlexVelezLl! confirming points 2 & 3 will be managed separately. I'll prioritize code review on this tomorrow, and Peter if you are able to re-review Alex's changes for items 1 and 4, that would be helpful. Thank you both

radinamatic commented:
No regressions or errors observed while navigating by keyboard through this new **Plan >Lesson summary** page. All interactive elements are focusable in a meaningful order, and actionable as expected.

However, 3 elements are missing the label (accessible text): 

Options button | Remove button | Visibility toggle 
-- | -- | --
<img width="840" alt="2024-10-24_14-10-20" src="https://github.com/user-attachments/assets/ebc53ab4-cb63-4723-9e18-20960633e880"> | <img width="840" alt="2024-10-24_14-12-45" src="https://github.com/user-attachments/assets/df43ffe3-7bd3-4107-b9b6-63654b442d90"> | <img width="840" alt="2024-10-24_14-13-35" src="https://github.com/user-attachments/assets/9ed1fc13-fa17-462e-9568-3985dc23d1f3">

radinamatic commented:
One thing is pre-existing, and we will report separately: the issue of the <kbd>UNDO</kbd> button inside the snackbar that appears when a resource is removed, and it needs to be put in focus immediately after the deletion.

AlexVelezLl committed:
Add missing aria labels

pcenov commented:
Hi @AlexVelezLl - I confirm that point 1 is fixed but for point 4 what I meant is that there should be more space between the label and the text below it in the side panel. Here's a screenshot:

![side panel](https://github.com/user-attachments/assets/29c96ffc-f4e4-468b-b0ed-7595dba7f374)

AlexVelezLl committed:
Fix status style

AlexVelezLl committed:
Fix status style

AlexVelezLl commented:
Thank you @radinamatic! I just added the aria labels for the options and remove buttons. For the switch one I will create a follow up in KDS as we wll need to fix this there (https://github.com/learningequality/kolibri-design-system/issues/806, https://github.com/learningequality/kolibri/issues/12743).

And @pcenov, I have updated the margins according to the specs:
![image](https://github.com/user-attachments/assets/a3903141-54fc-4d02-9a18-689ba87155b2)

radinamatic reviewed (approved):
Missing button labels added, margins according to specs, all issues addressed, and [follow-up issue](https://github.com/learningequality/kolibri-design-system/issues/806) filed! 👏🏽 

Excellent job, @AlexVelezLl! 💯 :shipit:

rtibbles commented on yarn.lock:
```diff
@@ -2963,15 +2963,6 @@ anymatch@^3.0.3, anymatch@~3.1.2:
     normalize-path "^3.0.0"
     picomatch "^2.0.4"
 
-"aphrodite@git+https://github.com/learningequality/aphrodite.git":
```
It seems like dependabot was responsible for adding this - I am not sure why it does!

AlexVelezLl replied:
Yes, I have the sense that we have had removed this a couple of times already :sweat_smile:/

rtibbles reviewed (commented):
I did a quick skim, and nothing jumped out at me, but it was by no means a comprehensive review.

marcellamaki commented on kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/index.vue:
```diff
@@ -207,6 +232,61 @@
+          },
+        );
+      },
+      async handleResourcesChange({ newArray }) {
```
All of the work here to handle the resource management within the table, while certainly implied by/part of the spec, is tough, and you did a great job with it. It's a lot more complicated than it would seem just by looking at the UI. And I really like the addition of the snackbar here and the `undo` action. It adds complexity but a really thoughtful implementation for the user perspective.

marcellamaki commented on kolibri/plugins/coach/assets/src/views/reports/ReportsLessonResourcesTable.vue:
```diff
@@ -127,22 +125,12 @@
     },
     methods: {
-      handleDragStart() {
-        // Used to mitigate the issue of text being selected while dragging
```
the upgrade to using `@mousedown.prevent` here is a small change but so nice

marcellamaki reviewed (approved):
Nice work @AlexVelezLl - this looks great, no concerns on my end, and some of the work here is just really thoughtful implementation and interpretation of the spec. Possibly you may end up wanting to make some further tweaks on this as the IA work wraps up, but that's fine with me, and we can continue any discussions about expected behavior as the work progresses. Nicely done!
